src/content/components/IntervalMarkerOverview.js

```diff
@@ -1,44 +1,189 @@
 import React, { Component, PropTypes } from 'react';
 import shallowCompare from 'react-addons-shallow-compare';
+import classNames from 'classnames';
+import { timeCode } from '../../common/time-code';
 import { withSize } from '../with-size';
 
+type MarkerState = 'PRESSED' | 'HOVERED' | 'NONE';
+
 class IntervalMarkerOverview extends Component {
   constructor(props) {
     super(props);
-    this._onSelect = this._onSelect.bind(this);
+    this.state = { hoveredItem: null, mouseDownItem: null };
+    this._onMouseDown = this._onMouseDown.bind(this);
+    this._onMouseMove = this._onMouseMove.bind(this);
+    this._onMouseUp = this._onMouseUp.bind(this);
+    this._onMouseOut = this._onMouseOut.bind(this);
+    this._canvas = null;
+    this._takeCanvasRef = c => { this._canvas = c; };
+  }
+
+  _scheduleDraw() {
+    if (!this._requestedAnimationFrame) {
+      this._requestedAnimationFrame = true;
+      window.requestAnimationFrame(() => {
+        this._requestedAnimationFrame = false;
+        if (this._canvas) {
+          timeCode('IntervalMarkerTimeline render', () => {
+            this.drawCanvas(this._canvas);
+          });
+        }
+      });
+    }
+  }
+
+  _hitTest(e) {
+    const r = this._canvas.getBoundingClientRect();
+    const { width, rangeStart, rangeEnd, intervalMarkers, styles } = this.props;
+    const x = e.pageX - r.left;
+    const y = e.pageY - r.top;
+    const time = rangeStart + x / width * (rangeEnd - rangeStart);
+
+    // Walk the list of intervalMarkers backwards to find the marker under the
+    // mouse. Later markers are drawn on top and we want to find the topmost one.
+    for (let i = intervalMarkers.length - 1; i >= 0; i--) {
```

Comment:
The comment tries to explain it: "Later markers are drawn on top and we want to find the topmost one." The word "later" is probably not the right one here - maybe "markers at a higher array index"? "subsequent"?